src/source/PeerConnection/Rtcp.c

```diff
@@ -7,26 +7,85 @@ STATUS onRtcpPacket(PKvsPeerConnection pKvsPeerConnection, PBYTE pBuff, UINT32 b
     STATUS retStatus = STATUS_SUCCESS;
     RtcpPacket rtcpPacket;
     UINT32 currentOffset = 0;
+    UINT64 currentTimeNTP = convertTimestampToNTP(GETTIME());
 
     CHK(pKvsPeerConnection != NULL && pBuff != NULL, STATUS_NULL_ARG);
 
     while (currentOffset < buffLen) {
         CHK_STATUS(setRtcpPacketFromBytes(pBuff + currentOffset, buffLen - currentOffset, &rtcpPacket));
 
-        if (rtcpPacket.header.packetType == RTCP_PACKET_TYPE_GENERIC_RTP_FEEDBACK && rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_NACK) {
-            CHK_STATUS(resendPacketOnNack(&rtcpPacket, pKvsPeerConnection));
-        } else if (rtcpPacket.header.packetType == RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK &&
-                   rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_APPLICATION_LAYER_FEEDBACK &&
-                   isRembPacket(rtcpPacket.payload, rtcpPacket.payloadLength) == STATUS_SUCCESS)
-        {
-            CHK_STATUS(onRtcpRembPacket(&rtcpPacket, pKvsPeerConnection));
-        } else if (rtcpPacket.header.packetType == RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK &&
-                   rtcpPacket.header.receptionReportCount == RTCP_PSFB_PLI)
-        {
-            CHK_STATUS(onRtcpPLIPacket(&rtcpPacket, pKvsPeerConnection));
+        switch (rtcpPacket.header.packetType) {
+            case RTCP_PACKET_TYPE_GENERIC_RTP_FEEDBACK:
+                if (rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_NACK)
+                    CHK_STATUS(resendPacketOnNack(&rtcpPacket, pKvsPeerConnection));
+                break;
+            case RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK:
+                if (rtcpPacket.header.receptionReportCount == RTCP_FEEDBACK_MESSAGE_TYPE_APPLICATION_LAYER_FEEDBACK &&
+                    isRembPacket(rtcpPacket.payload, rtcpPacket.payloadLength) == STATUS_SUCCESS) {
+                    CHK_STATUS(onRtcpRembPacket(&rtcpPacket, pKvsPeerConnection));
+                } else if (rtcpPacket.header.receptionReportCount == RTCP_PSFB_PLI) {
+                    CHK_STATUS(onRtcpPLIPacket(&rtcpPacket, pKvsPeerConnection));
+                } else {
+                    DLOGW("unhandled packet type RTCP_PACKET_TYPE_PAYLOAD_SPECIFIC_FEEDBACK");
+                }
+                break;
+            case RTCP_PACKET_TYPE_SENDER_REPORT: {
+                // https://tools.ietf.org/html/rfc3550#section-6.4.1
+                UINT32 x = rtcpPacket.payloadLength;
+                if (x == 24) {
+                    INT32 senderSSRC = getUnalignedInt32BigEndian(rtcpPacket.payload);
+                    UINT32 ntpHi = (UINT32) getUnalignedInt32BigEndian(rtcpPacket.payload + 4);
```

Comment:
Use SIZEOF(UINT32) as the offset instead of + 4, etc

Reply:
Also, why do you cast to UINT32? The fn has 32 bit return type

Reply:
getUnalignedInt32BigEndian returns unsigned INT32

Reply:
Oh, got it. Sorry, entirely forgot